### PR TITLE
cloud: support backup to s3 when object lock enabled.

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/13442

Problem Summary:

### What is changed and how it works?
1. wait for https://github.com/tikv/tikv/pull/13350/files merged.

### Check List

Tests 

- [ ] Manual test (add detailed scripts or steps below)
see https://github.com/tikv/tikv/pull/13350

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Support backup to s3 bucket when object lock enabled.
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]
LGTM

File: br/pkg/storage/s3.go [hunk 1]
```diff
@@ -387,11 +392,13 @@ func newS3Storage(backend *backuppb.S3, opts *ExternalStorageOptions) (obj *S3St
...
 	}
 
-	return &S3Storage{
+	s3Storage := &S3Storage{
 		session: ses,
 		svc:     c,
 		options: &qs,
-	}, nil
+	}
+	qs.ObjectLockEnabled = s3Storage.checkObjectLockEnabled()
```
Comment (YuJuncen) [inline br/pkg/storage/s3.go:400]: Why not just fill the `ObjectLockEnabled` directly into `s3Storage.ObjectLockEnabledEnabled`?

Comment (YuJuncen) [reply br/pkg/storage/s3.go:400]: (Well, it should be sent to TiKV, however should we modify the `kvproto` before doing that?)

**Review by Leavrth — APPROVED** [review]

**Comment by hawkingrei** [comment]
/hold 

@3pointer Your test case have timeout.

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**